server/datastore/mysql/software.go

```diff
@@ -2251,459 +2252,689 @@ func (ds *Datastore) ListCVEs(ctx context.Context, maxAge time.Duration) ([]flee
-		return nil, nil, fleet.NewInvalidArgumentError(
-			"query", "min_cvss_score, max_cvss_score, and exploit can only be provided with vulnerable=true",
-		)
+type hostSoftware struct {
+	fleet.HostSoftwareWithInstaller
+
+	LastInstallInstalledAt         *time.Time `db:"last_install_installed_at"`
+	LastInstallInstallUUID         *string    `db:"last_install_install_uuid"`
+	LastUninstallUninstalledAt     *time.Time `db:"last_uninstall_uninstalled_at"`
+	LastUninstallScriptExecutionID *string    `db:"last_uninstall_script_execution_id"`
+
+	ExitCode           *int       `db:"exit_code"`
+	LastOpenedAt       *time.Time `db:"last_opened_at"`
+	BundleIdentifier   *string    `db:"bundle_identifier"`
+	Version            *string    `db:"version"`
+	SoftwareID         *uint      `db:"software_id"`
+	SoftwareSource     *string    `db:"software_source"`
+	InstallerID        *uint      `db:"installer_id"`
+	PackageSelfService *bool      `db:"package_self_service"`
+	PackageName        *string    `db:"package_name"`
+	PackagePlatform    *string    `db:"package_platform"`
+	PackageVersion     *string    `db:"package_version"`
+	VPPAppSelfService  *bool      `db:"vpp_app_self_service"`
+	VPPAppAdamID       *string    `db:"vpp_app_adam_id"`
+	VPPAppVersion      *string    `db:"vpp_app_version"`
+	VPPAppPlatform     *string    `db:"vpp_app_platform"`
+	VPPAppIconURL      *string    `db:"vpp_app_icon_url"`
+
+	VulnerabilitiesList   *string `db:"vulnerabilities_list"`
+	SoftwareIDList        *string `db:"software_id_list"`
+	SoftwareSourceList    *string `db:"software_source_list"`
+	VersionList           *string `db:"version_list"`
+	BundleIdentifierList  *string `db:"bundle_identifier_list"`
+	VPPAppSelfServiceList *string `db:"vpp_app_self_service_list"`
+	VPPAppAdamIDList      *string `db:"vpp_app_adam_id_list"`
+	VPPAppVersionList     *string `db:"vpp_app_version_list"`
+	VPPAppPlatformList    *string `db:"vpp_app_platform_list"`
+	VPPAppIconUrlList     *string `db:"vpp_app_icon_url_list"`
+}
+
+func hostInstalledSoftware(ds *Datastore, ctx context.Context, hostID uint) ([]*hostSoftware, error) {
+	installedSoftwareStmt := `
+		SELECT
+			software_titles.id AS id,
+			host_software.software_id AS software_id,
+			host_software.last_opened_at,
+			NULL AS status
+		FROM 
+			host_software
+		INNER JOIN
+			software ON host_software.software_id = software.id
+		INNER JOIN
+			software_titles ON software.title_id = software_titles.id
+		WHERE
+			host_software.host_id = :host_id
+	`
+	installedSoftwareStmt, args, err := sqlx.Named(installedSoftwareStmt, map[string]any{
+		"host_id": hostID,
+	})
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "build named query for host installed software")
 	}
 
-	var onlySelfServiceClause string
-	if opts.SelfServiceOnly {
-		onlySelfServiceClause = ` AND ( si.self_service = 1 OR ( vat.self_service = 1 AND :is_mdm_enrolled ) ) `
+	installedSoftwareStmt = ds.reader(ctx).Rebind(installedSoftwareStmt)
```

Comment:
Thinking we don't need the rebinds here; in most places (including the other queries in this diff) we don't have them, and we're not going to swap to Postgres or the like any time soon.